### PR TITLE
Fix various local env issues and update the README

### PR DESCRIPTION
**Pull request by sarayourfriend**

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Not sure why the CI isn't running 🤔 Probably some kind of yaml syntax error?

**Comment by sarayourfriend** [comment]
Never mind, I had duplicate keys. Figured it out 😅

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Test
```
Comment (AetherUnbound) [inline .github/workflows/pull_request.yml]: I might rename this `Lint and Test` - also, would it be okay to include `black` and `flake8` here as well? Those are static checks that don't need to run within the docker-compose setup, they can run on the code directly. (I think `mypy` can also run staticly w/o an environment but don't quote me on that)

Comment (sarayourfriend) [reply .github/workflows/pull_request.yml]: Yeah actually we could just move those to a pre-commit hook 😀

Comment (sarayourfriend) [reply .github/workflows/pull_request.yml]: > I think mypy can also run staticly w/o an environment but don't quote me on that

I tried to get it to run but inside pre-commit it required too many dependencies and then environment variables that didn't exist. It seemed to make more sense to just run it inside the container where it already works.

Comment (AetherUnbound) [reply .github/workflows/pull_request.yml]: Gah, okay I'm remembering now that one of my work project runs mypy inside the container too 😕 ah well, thanks for trying that and adding the pre-commit file!

**Review by AetherUnbound — APPROVED** [review]
🟢

**Pull request merged**